lib/src/atom/matcher.rs

```diff
@@ -897,7 +914,8 @@ pub fn apply_bindings_to_atom(atom: &Atom, bindings: &Bindings) -> Atom {
 pub fn apply_bindings_to_bindings(from: &Bindings, to: &Bindings) -> Result<Bindings, ()> {
     // TODO: apply_bindings_to_bindings can be replaced by Bindings::merge,
     // when Bindings::merge are modified to return Vec<Bindings>
-    Bindings::merge(to, from).filter(|bindings| !bindings.has_loops()).ok_or(())
+    //LP QUESTION: unclear whether behavior to filter out Bindings with loops continues to make this function relevant
```

Comment:
No, filtering loops doesn't make this function unique. In fact I mechanically moved this check from a previous version which implemented custom algorithm and this check was needed. Now we have `Bindings::merge` and all these checks are inside. So we can remove this function safely.

Reply:
Ok.  I eliminated the call from GroundingSpace::query, as it appeared redundant.  I did not touch Interpreter.